lib/menubar/utils.lua

```diff
@@ -343,8 +343,13 @@ function utils.parse_desktop_file(file)
             cmdline = cmdline:gsub('%%i', '')
         end
         if program.Terminal == true then
+            program.with_shell = true
             cmdline = utils.terminal .. ' -e ' .. cmdline
         end
+        if program.Path then
+            program.with_shell = true
+            cmdline = 'cd ' .. program.Path .. ' && ' .. cmdline
```

Comment:
To avoid problems with spaces, path can just be quoted like this as @sclu1034 suggested
```suggestion
            cmdline = 'cd "' .. program.Path .. '" && ' .. cmdline
```